src/assert.c

```diff
@@ -420,6 +441,20 @@ fido_verify_sig_es256(const fido_blob_t *dgst, const es256_pk_t *pk,
 		return (-1);
 	}
 
+#if OPENSSL_VERSION_NUMBER >= 0x30000000
+	if ((pkey = es256_pk_to_EVP_PKEY(pk)) == NULL ||
+	    (pctx = EVP_PKEY_CTX_new(pkey, NULL)) == NULL) {
+		fido_log_debug("%s: pk -> ec", __func__);
+		goto fail;
+	}
+
+	if (EVP_PKEY_verify_init(pctx) != 1 ||
+	    EVP_PKEY_verify(pctx, sig->ptr, sig->len,
+	    dgst->ptr, dgst->len) != 1) {
+		fido_log_debug("%s: EVP_PKEY_verify", __func__);
+		goto fail;
+	}
```

Comment:
A similar approach could be taken here (i.e, switch to `EVP_PKEY` and drop the `#if`).